### DISCUSSION TITLE
Why OSC track id shows `sid/aid` after mpv v0.39.0

### DISCUSSION THREAD

**Discussion by verygoodlee** [mpv-player/mpv#15434] [General]
In v0.38.0 shows OSC counted id.
https://github.com/mpv-player/mpv/blob/02254b92dd237f03aa0a151c2a68778c4ea848f9/player/lua/osc.lua#L426-L436
this function was removed after v0.39.0 and replaced with the sid/aid prop.
https://github.com/mpv-player/mpv/blob/78ee714fff5f21d91d996eab0a1a06f622d6cb0b/player/lua/osc.lua#L1873-L1879

sid/aid may greater than track count, resulting in this weird track information.
![image](https://github.com/user-attachments/assets/a50fb13a-a180-49e9-ab66-3e66d9b4a9b2)
occurs after sid 6 removed by sub-remove command.
![image](https://github.com/user-attachments/assets/179060bc-386d-4291-8dc0-c999da4d38fe)

**Comment by verygoodlee** [comment]
@guidocella Introduced in mpv-player/mpv@1a558bf5

this statement is wrong, sub-remove/audio-remove command might make them unequal.
> mpv's track id and the osc_id are exactly the same

**Comment by guidocella** [comment]
It uses `sid` and `aid` to reduce the size of the code. It is not worth keeping track of removed tracks. If anything the track id should not be shown at all since it is not useful information. The track button can be drawn gray to show that no track is selected, and be hidden to show that no tracks are available instead.

**Reply by verygoodlee** [reply]
> The track button can be drawn gray to show that no track is selected, and be hidden to show that no tracks are available instead.

gray usually means button is unclickable, which is more suitable for indicating that there is no available tracks, 
may need to design another icon to indicate that no track is selected, something like this ![屏幕截图 2024-12-04 150621](https://github.com/user-attachments/assets/0bdc3a52-f268-474a-b31a-e61934812ff3)

**Reply by guidocella** [reply]
Yeah, even better. Also note that using different ids caused a different bug in https://github.com/mpv-player/mpv/issues/10968
